elementary/monitor/api/tests/tests.py

```diff
@@ -87,222 +74,227 @@ def _get_invocation_from_filter(
     ]:
-        test_results_metadata = self.get_run_cache(TESTS_METADATA)
+        filtered_test_results_db_rows = test_results_db_rows
         invocation = self._get_invocation_from_filter(filter)
         if invocation.invocation_id:
-            test_results_metadata = self.get_tests_metadata(
-                invocation_id=invocation.invocation_id, should_cache=False
-            )
-        elif test_results_metadata is None:
-            test_results_metadata = self.get_tests_metadata(days_back=days_back)
+            filtered_test_results_db_rows = [
+                test_result
+                for test_result in filtered_test_results_db_rows
+                if test_result.invocation_id == invocation.invocation_id
+            ]
 
-        tests_sample_data = self.get_run_cache(TESTS_SAMPLE_DATA)
-        if tests_sample_data is None:
-            tests_sample_data = self.get_tests_sample_data(
-                days_back=days_back,
-                metrics_sample_limit=metrics_sample_limit,
-                disable_passed_test_metrics=disable_passed_test_metrics,
-                disable_samples=disable_samples,
-            )
+        filtered_test_results_db_rows = [
+            test_result
+            for test_result in filtered_test_results_db_rows
+            if test_result.invocations_order == 1
+        ]
 
         test_results = defaultdict(list)
-        for test_metadata in test_results_metadata:
-            test_metadata_dict = dict(test_metadata)
-            test_sub_type_unique_id = self.get_test_sub_type_unique_id(
-                **test_metadata_dict
-            )
-            test_sample_data = tests_sample_data.get(test_sub_type_unique_id)
+        for test_result_db_row in filtered_test_results_db_rows:
             test_result = TestResultSchema(
-                metadata=self.get_test_info_from_test_metadata(test_metadata),
-                test_results=test_metadata.get_test_results(test_sample_data).dict(),
+                metadata=self.get_test_metadata_from_test_result_db_row(
+                    test_result_db_row
+                ),
+                test_results=TestsAPI._get_test_result_from_test_result_db_row(
+                    test_result_db_row, disable_samples=disable_samples
+                ),
             )
-            test_results[test_metadata.model_unique_id].append(test_result)
+            test_results[test_result_db_row.model_unique_id].append(test_result)
 
-        self.set_run_cache(key=TEST_RESULTS, value=test_results)
         return test_results, invocation
 
+    @staticmethod
+    def _get_failed_rows_count(test_result_db_row: TestResultDBRowSchema) -> int:
+        failed_rows_count = -1
+        if (
+            test_result_db_row.status != "pass"
+            and test_result_db_row.test_results_description
+        ):
+            found_rows_number = re.search(
+                r"\d+", test_result_db_row.test_results_description
```

Comment:
Why don't we use the `failures` column?

Reply:
This is an old code when snowflake didn't return the real failures (affected rows).
So we can change it and use it, but that means that users that are using old snowflake adapter won't have a real number (only 1)